### ISSUE TITLE
Document how to fall back to normal Java/Win/Linux menus

### ISSUE THREAD

**Issue by msgilligan** [codecentric/NSMenuFX#13]
How do I write an app that will work on Windows and Linux? It would be nice if there was some documentation or an example of how to fall back to "standard" behavior on systems that are not Mac OS X.

**Comment by msgilligan** [comment]
With the current version it appears `MenuToolkit.toolkit()` returns `null` on non-MacOSX systems. I was hoping I'd be able to still have a Windows menu, etc. on the other platforms.

**Comment by 0x4a616e** [comment]
That is currently not so nice, indeed. Maybe I can create some kind of "fallback" toolkit for other platforms that allows you to use at least the window menu... Other things like the "Hide Others", "Show All" menu items will however not be available on other platforms, so I have to think about some fallback solution there and a general interface for MenuToolkit that can be available on all platforms.

**Comment by msgilligan** [comment]
Yeah, I'm thinking that might make sense in some cases. Especially when you are developing on the Mac and don't have much time to test/tweak on the other systems.

In cases where you're making a serious cross-platform app you probably want Windows or Linux-specific integration and don't want to look too much like something originally intended for the Mac.

I think there are two different usage scenarios.Right now, for example, I'm in the early stages of making a a Mac app and would be happy to have consistent (though Mac-like) behavior on the other platforms. Later on I'd probably want to focus on looking more native on Windows and Linux.

**Comment by 0x4a616e** [comment]
I added a new fallback mechanism in the last major update. See https://github.com/codecentric/NSMenuFX/issues/36 for more details. You'll now get a fallback toolkit on other platforms which tries to support as much as possible. The API is not carved in stone yet, so feel free to provide some feedback! :)
